templates/single-product/related.php

```diff
@@ -46,9 +46,9 @@
 
 if ( $products->have_posts() ) : ?>
 
-	<div class="related products">
+	<section class="related products">
 
-		<h2><?php _e( 'Related Products', 'woocommerce' ); ?></h2>
+		<h2><?php echo sprintf( __( 'Other %s...', 'woocommerce' ), $product->get_categories() ); ?></h2>
```

Comment:
> The only alternative I can think of would be to check how many categories the product is in and if it's less than 5 (or so) then display them in this heading, otherwise just display the generic 'related products' heading.
 
What if a product of total different category is displaying in related because of its tag...
As Claudio wisely suggested, we should better forget about it...